python/examples/calendar/schema.py

```diff
@@ -0,0 +1,89 @@
+from typing import Literal, NotRequired, TypedDict, Annotated
+def Doc(s: str) -> str: return s
+
+
+class UnknownAction(TypedDict):
+    """
+    if the user types text that can not easily be understood as a calendar action, this action is used
+    """
+
+    actionType: Literal["Unknown"]
+    text: Annotated[str, Doc("text typed by the user that the system did not understand")]
+
+
+class EventTimeRange(TypedDict):
```

Comment:
I would add `total=True` and drop all the `NotRequired`

See https://docs.python.org/3/library/typing.html#typing.TypedDict